src/Widgets/Stopwatch.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2019-2021 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+#include "Widgets/Stopwatch.hpp"
+#include "generated/SettingsHelper.hpp"
+
+#include <QCoreApplication>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QPushButton>
+#include <QString>
+#include <QTime>
+#include <QTimer>
+
+namespace Widgets
+{
+
+Stopwatch::Stopwatch(QWidget *parent) : QWidget(parent)
+{
+    // construct widgets
+    mainLayout = new QHBoxLayout(this);
+    timeLabel = new QLabel("00:00:00");
+    timeLabel->setAlignment(Qt::AlignHCenter | Qt::AlignVCenter);
+    startPauseButton = new QPushButton("Start");
+    resetButton = new QPushButton("Reset");
+
+    // set up UI
+    mainLayout->addWidget(timeLabel);
+    mainLayout->addWidget(startPauseButton);
+    mainLayout->addWidget(resetButton);
+
+    // set up timer
+    timer = new QTimer(this);
+    timer->setInterval(getGranularity());
+    timer->callOnTimeout(this, &Stopwatch::update);
+
+    connect(startPauseButton, &QPushButton::clicked, this, &Stopwatch::startOrPauseStopwatch);
+    connect(resetButton, &QPushButton::clicked, this, &Stopwatch::resetStopwatch);
+}
+
+void Stopwatch::startOrPauseStopwatch()
+{
+    isRunning() ? pause() : start();
+    update();
+}
+
+void Stopwatch::resetStopwatch()
+{
+    reset();
+    update();
+}
+
+void Stopwatch::start()
+{
+    if (isInactive())
+    {
+        accumulator = 0;
+        elapsedTimer.start();
+        startTimer(getGranularity());
```

Comment:
What does this do :thinking:

Reply:
Sorry. I forgot to remove it. I was trying out something. Not needed now.